### PR TITLE
[Template-Message] Feat/Add Template Message models

### PR DESCRIPTION
**Pull request by pauloabreu**
# Resume
This PR add Template Message models to support new features

### CODE REVIEW

File: marketplace/wpp_templates/models.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        ("IMAGE", ""),
+        ("DOCUMENT", ""),
+        ("VIDEO", ""),
+    )
+
+    header_type = models.CharField(max_length=20, choices=HEADER_TYPE_CHOICES)
+    text = models.CharField(max_length=60)
+
+    def save(self, **kwargs):
+        if self.header_type == "TEXT" and not self.text:
```
Comment (Sandro-Meireles) [inline marketplace/wpp_templates/models.py:114]: This can be done using constraints, so it's safer:
https://docs.djangoproject.com/en/4.1/ref/models/constraints/

File: marketplace/wpp_templates/models.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
+import uuid
+
+from django.db import models
+
+from rest_framework.exceptions import ValidationError
+
+
+class TemplateMessage(models.Model):
```
Comment (Sandro-Meireles) [inline marketplace/wpp_templates/models.py:8]: In the planning, the name of the model was `Template`, what is the reason for this change?

File: marketplace/wpp_templates/models.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    uuid = models.UUIDField(default=uuid.uuid4, editable=False, unique=True)
+    name = models.CharField(max_length=60)
+
+    CATEGORY_CHOICES = (
```
Comment (Sandro-Meireles) [inline marketplace/wpp_templates/models.py:12]: Constants must be at the top of the class
this works for such lines:
- [32](https://github.com/weni-ai/weni-integrations-engine/pull/166/files#diff-73c4599f012834487ca0855b3d039688b8318535f13ad788afc59f9fab46c419R32)
- [48](https://github.com/weni-ai/weni-integrations-engine/pull/166/files#diff-73c4599f012834487ca0855b3d039688b8318535f13ad788afc59f9fab46c419R48)
- [77](https://github.com/weni-ai/weni-integrations-engine/pull/166/files#diff-73c4599f012834487ca0855b3d039688b8318535f13ad788afc59f9fab46c419R77)
- [103](https://github.com/weni-ai/weni-integrations-engine/pull/166/files#diff-73c4599f012834487ca0855b3d039688b8318535f13ad788afc59f9fab46c419R103)

File: marketplace/wpp_templates/models.py (hunk 2 above)
Comment (Sandro-Meireles) [inline marketplace/wpp_templates/models.py:8]: This model relates to the App
image

File: marketplace/wpp_templates/models.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    language = models.CharField(max_length=60)
+    country = models.CharField(max_length=60)
+    namespace = models.CharField(max_length=60)
+    external_id = models.CharField(max_length=60)
+
+
+class TemplateButton(models.Model):
+    uuid = models.UUIDField(default=uuid.uuid4, editable=False, unique=True)
+
+    template_translation = models.ForeignKey(TemplateTranslation, on_delete=models.PROTECT)
```
Comment (Sandro-Meireles) [inline marketplace/wpp_templates/models.py:75]: The name can just be `translation`

File: marketplace/wpp_templates/models.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        ("IN_APPEAL", ""),
+        ("PENDING", ""),
+        ("REJECTED", ""),
+        ("PENDING_DELETION", ""),
+        ("DELETED", ""),
+        ("DISABLED", ""),
+        ("LOCKED", ""),
+    )
+
+    template = models.ForeignKey(TemplateMessage, on_delete=models.PROTECT, related_name="template_translations")
```
Comment (Sandro-Meireles) [inline marketplace/wpp_templates/models.py:59]: The related_name can just be `translations` like: https://github.com/Ilhasoft/rapidpro/blob/main/temba/templates/models.py#L69

File: marketplace/wpp_templates/models.py (hunk 4 above)
Comment (Sandro-Meireles) [inline marketplace/wpp_templates/models.py:75]: Add a related name: `related_name="buttons"`

File: marketplace/wpp_templates/models.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+        if self.button_type == "PHONE_NUMBER" and not self.phone_number:
+            raise ValidationError("phone_number field is required for PHONE_NUMBER button_type")
+        super().save(**kwargs)
+
+
+class TemplateHeader(models.Model):
+    uuid = models.UUIDField(default=uuid.uuid4, editable=False, unique=True)
+
+    template_translation = models.ForeignKey(TemplateTranslation, on_delete=models.PROTECT)
```
Comment (Sandro-Meireles) [inline marketplace/wpp_templates/models.py:101]: The name can just be `translation`

Comment (Sandro-Meireles) [inline marketplace/wpp_templates/models.py:101]: Add a related name: `related_name="headers"`

**Comment by Sandro-Meireles** [comment]
Upload migrations along with ess PR

File: marketplace/wpp_templates/models.py (hunk 2 above)
Comment (pauloabreu) [reply marketplace/wpp_templates/models.py:8]: I think the name Template might be confused
